src/components/Dialogs/ImageCropDialog/cropper.ts

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react'
+import Cropper from 'cropperjs'
+import 'cropperjs/dist/cropper.min.css'
+
+const INITIAL_ZOOM = 0.1
+
+export type CropperImageType = 'avatar' | 'videoThumbnail' | 'cover'
+
+type UseCropperOpts = {
+  imageEl: HTMLImageElement | null
+  imageType: CropperImageType
+}
+
+const ASPECT_RATIO_PER_TYPE: Record<CropperImageType, number> = {
+  avatar: 1,
+  videoThumbnail: 16 / 9,
+  cover: 4,
+}
+
+const CANVAS_OPTS_PER_TYPE: Record<CropperImageType, Cropper.GetCroppedCanvasOptions> = {
+  avatar: {
+    minWidth: 128,
+    minHeight: 128,
+    width: 256,
+    height: 256,
+    maxWidth: 1024,
+    maxHeight: 1024,
+  },
+  videoThumbnail: {
+    minWidth: 1280,
+    minHeight: 720,
+    width: 1920,
+    height: 1080,
+    maxWidth: 3840,
+    maxHeight: 2160,
+  },
+  cover: {
+    minWidth: 1920,
+    minHeight: 480,
+    width: 1920,
+    height: 480,
+    maxWidth: 3840,
+    maxHeight: 960,
+  },
+}
+
+export const useCropper = ({ imageEl, imageType }: UseCropperOpts) => {
+  const [cropper, setCropper] = useState<Cropper | null>(null)
+  const [currentZoom, setCurrentZoom] = useState(INITIAL_ZOOM)
+
+  const handleZoomChange = (zoom: number) => {
+    setCurrentZoom(zoom)
+    cropper?.zoomTo(zoom)
+  }
+
+  // initialize
+  useEffect(() => {
+    if (!imageEl) {
+      return
+    }
+
+    const handleZoomEvent = (event: Cropper.ZoomEvent<HTMLImageElement>) => {
+      const { ratio, oldRatio } = event.detail
+      console.log({ ratio, oldRatio })
+      // TODO fix min zoom
```

Comment:
This still needs adjustment, it's not very straightforward though. Will fix tomorrow